storage/store/inmemory/unsynchronized/transaction_result_error_messages.go

```diff
@@ -125,3 +127,25 @@ func (t *TransactionResultErrorMessages) Store(
 
 	return nil
 }
+
+// AddToBatch adds all the in-memory storages to the given batch.
+// It is used for the batching writes to the DB.
+func (t *TransactionResultErrorMessages) AddToBatch(batch storage.ReaderBatchWriter) error {
+	writer := batch.Writer()
+
+	for block, errorMessages := range t.blockStore {
+		decoded, err := store.KeyToBlockID(block)
+		if err != nil {
+			return fmt.Errorf("could not decode block: %w", err)
+		}
+
+		for _, msg := range errorMessages {
```

Comment:
this also needs to index
https://github.com/onflow/flow-go/blob/0276508117497cb2ea62877b71526d2d5bf78269/storage/store/transaction_result_error_messages.go#L119-L129